src/test/java/org/opensearch/neuralsearch/search/collector/HybridCollapsingTopDocsCollectorTests.java

```diff
@@ -0,0 +1,226 @@
+    private static final String COLLAPSE_FIELD_NAME = "collapseField";
+    private static final String TEST_QUERY_TEXT = "greeting";
+    private static final String TEST_QUERY_TEXT2 = "salute";
+    private static final int NUM_DOCS = 4;
+    private static final int TOP_N_GROUPS = 2;
+    private static final int TOTAL_HITS_UP_TO = 1000;
+
+    public void testKeywordCollapse_whenCollectAndTopDocs_thenSuccessful() throws IOException {
+        Directory directory = newDirectory();
+        IndexWriter writer = new IndexWriter(directory, newIndexWriterConfig(new MockAnalyzer(random())));
+
+        addKeywordDoc(writer, 0, "text1", 100, "group1");
+        addKeywordDoc(writer, 1, "text2", 200, "group1");
+        addKeywordDoc(writer, 2, "text3", 300, "group2");
+        addKeywordDoc(writer, 3, "text4", 400, "group2");
+        writer.commit();
+
+        DirectoryReader reader = DirectoryReader.open(writer);
+        LeafReaderContext context = reader.leaves().get(0);
+
+        Sort sort = new Sort(SortField.FIELD_SCORE);
+        KeywordFieldMapper.KeywordFieldType fieldType = new KeywordFieldMapper.KeywordFieldType(COLLAPSE_FIELD_NAME);
+
+        HybridCollapsingTopDocsCollector<?> collector = HybridCollapsingTopDocsCollector.createKeyword(
+            COLLAPSE_FIELD_NAME,
+            fieldType,
+            sort,
+            TOP_N_GROUPS,
+            new HitsThresholdChecker(TOTAL_HITS_UP_TO)
+        );
+
+        Weight weight = mock(Weight.class);
+        collector.setWeight(weight);
+        LeafCollector leafCollector = collector.getLeafCollector(context);
+
+        int[] docIds = new int[] { 0, 1, 2, 3 };
+        List<Float> scores1 = Arrays.asList(0.5f, 0.7f, 0.3f, 0.9f);
+
+        HybridSubQueryScorer hybridScorer = new HybridSubQueryScorer(1);
+
+        leafCollector.setScorer(hybridScorer);
+
+        collectDocsAndScores(hybridScorer, scores1, leafCollector, 0, docIds);
+
+        List<CollapseTopFieldDocs> topDocs = collector.topDocs();
+
+        assertEquals(1, topDocs.size());  // One for each sub-query
+
+        for (int i = 0; i < topDocs.size(); i++) {
+            CollapseTopFieldDocs collapseTopFieldDocs = topDocs.get(i);
+            assertEquals(4, collapseTopFieldDocs.totalHits.value());
+            assertEquals(4, collapseTopFieldDocs.scoreDocs.length);
+
+            // Verify collapse values
+            Set<String> uniqueGroups = new HashSet<>();
+            for (Object collapseValue : collapseTopFieldDocs.collapseValues) {
+                uniqueGroups.add(((BytesRef) collapseValue).utf8ToString());
+            }
+            assertEquals(2, uniqueGroups.size());
+            assertTrue(uniqueGroups.contains("group1"));
+            assertTrue(uniqueGroups.contains("group2"));
+        }
+
+        reader.close();
+        writer.close();
+        directory.close();
+    }
+
+    public void testNumericCollapse_whenCollectAndTopDocs_thenSuccessful() throws IOException {
```

Comment:
Write a test with and without concurrent segement search enabled.

Reply:
Should that be an integTest?